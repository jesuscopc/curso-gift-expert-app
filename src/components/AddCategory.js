import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategorie }) => {

  const [inputValue, setInputValue] = useState('');
  
  const handleChangeInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputValue.trim().length > 2) {
      setCategorie( [inputValue] );
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <h1>{ inputValue }</h1>
      <input
        type="text"
        value={ inputValue }
        onChange={ handleChangeInputValue }
      />
    </form>
  )
};

AddCategory.propTypes = {
  setCategorie: PropTypes.func.isRequired,
};