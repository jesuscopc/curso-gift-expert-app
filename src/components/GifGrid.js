import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categorie }) => {

  const { data: imgs, loading } = useFetchGifs( categorie );

  return (
    <>
      <h3 className="card animate__animated animate__fadeIn">{ categorie }</h3>
      { loading && <p className="card animate__animated animate__flash">loading...</p> }
      <div className="card-grid">
          {imgs.map( img =>
            (
              <GifGridItem
                key={ img.id }
                { ...img } 
              />
            )
          )}
      </div>
    </>
  )
}
