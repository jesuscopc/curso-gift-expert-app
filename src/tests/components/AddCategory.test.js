import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Sui test AddCategoryComponent', () => {
  
  const setCategorie = jest.fn();
  let wrapper = shallow(<AddCategory setCategorie={setCategorie}/>);

  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategorie={setCategorie}/>);
  });

  test('should be render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should be change the textinput value', () => {
    const input = wrapper.find('input');
    const value = 'Texto cambiado'
    input.simulate('change', { target: { value: value}});
    expect(wrapper.find('input').props().value.trim()).toBe(value);
  });

  test('should be not send data form', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} } );
    expect(setCategorie).not.toHaveBeenCalled();
  })
  
  test('should be submit the form and reset the input text', () => {
    const value = 'Hunter x Hunter';
    const input = wrapper.find('input').simulate('change',{ target: { value }});
    wrapper.find('form').simulate('submit',{ preventDefault(){} });
    expect(setCategorie).toHaveBeenLastCalledWith( expect.any(Array));
    expect(input.props().value.trim()).toBe('');
  })
  
  
  
})
