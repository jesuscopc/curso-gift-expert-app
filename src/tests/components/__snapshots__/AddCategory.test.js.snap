// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sui test AddCategoryComponent should be render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1 />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
